File: core/autonomy/semantic_planning_engine.py
from __future__ import annotations

from typing import Any, Dict, List

from .semantic_goal_engine import resolve_semantic_goal
from .semantic_task_decomposition_engine import decompose_semantic_task


MAX_PLAN_STEPS = 1000


def plan_semantic_autonomy(
    payload: Dict[str, Any],
) -> Dict[str, Any]:
    goal = resolve_semantic_goal(payload)
    decomposition = decompose_semantic_task(goal)
    steps = [
        {
            "step": idx,
            "action": subtask.get("semantic_unit"),
            "task_id": subtask.get("id"),
        }
        for idx, subtask in enumerate(
            decomposition.get("subtasks", [])[:MAX_PLAN_STEPS]
        )
    ]
    return {
        "goal": goal,
        "steps": steps,
        "step_count": len(steps),
        "bounded": True,
    }

File: src/autonomy/semanticPlanningEngine.ts
/**
 * Converted from Python: core/autonomy/semantic_planning_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { resolveSemanticGoal } from "./semanticGoalEngine.js";
import { decomposeSemanticTask } from "./semanticTaskDecompositionEngine.js";

export let MAX_PLAN_STEPS: any = 1000;
export function planSemanticAutonomy(payload: any): any {
  var goal: any = resolveSemanticGoal(payload);
  var decomposition: any = decomposeSemanticTask(goal);
  var steps: any = py.enumerate(py.slice(py.get(decomposition, "subtasks", []), null, MAX_PLAN_STEPS)).map(([idx, subtask]: any) => ({"step": idx, "action": py.get(subtask, "semantic_unit"), "task_id": py.get(subtask, "id")}));
  return {"goal": goal, "steps": steps, "step_count": py.len(steps), "bounded": true};
}
export { decomposeSemanticTask, resolveSemanticGoal };
